From 52426acb30ad462e6ceb9c0553dd30b192fd6eab Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Fri, 21 Jun 2024 18:17:44 +0100
Subject: [PATCH] arch: arm64: broadcom: Add firmware clocks node

BCM2712 still uses the firmware clocks driver, so add this to the
base device tree.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 .../boot/dts/broadcom/bcm2712-rpi-5-b.dts     | 28 +++++++++++++++++++
 1 file changed, 28 insertions(+)

--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
@@ -706,3 +706,31 @@ spi10_cs_pins: &spi10_cs_gpio1 {};
 		sd_cqe = <&sdio1>, "supports-cqe?";
 	};
 };
+
+&soc {
+	firmware: firmware {
+		compatible = "raspberrypi,bcm2835-firmware", "simple-mfd";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		mboxes = <&mailbox>;
+		dma-ranges;
+
+		firmware_clocks: clocks {
+			compatible = "raspberrypi,firmware-clocks";
+			#clock-cells = <1>;
+		};
+
+		reset: reset {
+			compatible = "raspberrypi,firmware-reset";
+			#reset-cells = <1>;
+		};
+	};
+
+	power: power {
+		compatible = "raspberrypi,bcm2835-power";
+		firmware = <&firmware>;
+		#power-domain-cells = <1>;
+	};
+
+};
